xcube/core/store/fs/impl/dataset.py

```diff
@@ -359,29 +361,57 @@ def open_data(self,
             file_path = protocol + "://" + data_id
         tile_size = open_params.get("tile_size", (512, 512))
         overview_level = open_params.get("overview_level", None)
-        return self.open_dataset(file_path, tile_size,
+        return self.open_dataset(fs, file_path, tile_size,
                                  overview_level=overview_level)
 
+    @classmethod
+    def create_env_session(cls, fs):
+        session = rasterio.env.Env(aws_secret_access_key=fs.token,
+                                   aws_access_key_id=fs.key,
+                                   aws_session_token=fs.token,
+                                   region_name=fs.client_kwargs.get(
+                                       'region_name',
+                                       'eu-central-1'
+                                   ),
+                                   AWS_NO_SIGN_REQUEST=fs.anon if fs.anon
+                                   else True)
+        return session
+
+    @classmethod
+    def open_dataset_with_rioxarray(cls, file_path, overview_level, tile_size):
+        return rioxarray.open_rasterio(
+            file_path,
+            overview_level=overview_level,
+            chunks=dict(zip(('x', 'y'), tile_size))
+        )
+
     @classmethod
     def open_dataset(cls,
-                     file_spec: str,
+                     file_spec,
```

Comment:
Why name `file_spec`? The type is obviously `Optional[fsspec.AbstractFileSystem]` so why not call it `fs` or `file_system`?